crates/ruff_linter/src/rules/perflint/rules/manual_list_comprehension.rs

```diff
@@ -219,6 +218,34 @@ pub(crate) fn manual_list_comprehension(checker: &mut Checker, for_stmt: &ast::S
         return;
     }
 
+    // Avoid if the for-loop target is used outside of the for loop, e.g.,
+    //
+    // ```python
+    // for x in y:
+    //     filtered.append(x)
+    // print(x)
+    // ```
+    //
+    // If this were a comprehension, x would no longer have the correct scope:
+    //
+    // ```python
+    // filtered = [x for x in y]
+    // print(x)
+    // ```
+    let for_loop_target = checker
+        .semantic()
+        .lookup_symbol(id.as_str())
```

Comment:
For some reason, `resolve_name` returns `None` for the for-loop target. Also, references to it outside of the for-loop are not included in its list of references; e.g.
```python
def f():
    result = []
    for val in range(5):
        result.append(val * 2)
    print(val) # this reference is not included in the list of references
```

Reply:
I think the problem here is that we build the semantic model lazily and it hasn't reached the point after the for loop yet

Reply:
@charliermarsh any suggestions on how to best find all usages of target?

Reply:
I think a working version would look something like this:

```rust
let for_loop_read = checker.semantic().resolve_load(for_loop_target_name_expr);
let for_loop_binding = match for_loop_read {
    ReadResult::Resolved(id) => checker.semantic().binding(id),
    _ => unreachable!("for loop target must exist"),
};
```

Unfortunately, `resolve_load` requires an `&mut SemanticModel` and I don't see a way to get one from the checker. Am I missing something here?

Reply:
Sorry, I didn't look deeply, but often in those cases you need to make it a deferred rule, so it runs _after_ the semantic model has been built (e.g., run it in `deferred_for_loops`).